consensus/core/src/linearizer.rs

```diff
@@ -177,9 +178,13 @@ impl Linearizer {
     }
 
     // This function should be called whenever a new commit is observed. This will
-    // iterate over the sequence of committed leaders and produce a list of sub-dags.
-    pub fn handle_commit(&mut self, committed_leaders: Vec<VerifiedBlock>) -> Vec<CommittedSubDag> {
-        let mut committed_sub_dags = vec![];
+    // iterate over the sequence of committed leaders and produce a list of commit
+    // data & committed sub-dags.
+    pub fn handle_commit(
+        &mut self,
+        committed_leaders: Vec<VerifiedBlock>,
+    ) -> Vec<(Commit, CommittedSubDag)> {
```

Comment:
If Commit is returned only for persistence, we can revert the change and ignore the persistence logic in this PR. Commit can be buffered in DagState, which needs access to the last Commit as well.

Reply:
This sounds good for now.